src/components/datePicker/datePicker.tsx

```diff
@@ -0,0 +1,127 @@
+import ReactDatePicker, { registerLocale } from 'react-datepicker';
+import classNames from 'classnames/bind';
+import { FC, ReactNode, useMemo, useRef, forwardRef } from 'react';
+import { be, ru, enGB, es, uk, zhCN, Locale } from 'date-fns/locale';
+import { DatePickerHeader } from './header/datePickerHeader';
+import Input from './input/input';
+import styles from './datePicker.module.scss';
+
+const cx = classNames.bind(styles);
+
+const currentLanguageToLocale: { [key: string]: Locale } = {
+  en: enGB,
+  es,
+  be,
+  ru,
+  uk,
+  zh: zhCN,
+};
+
+const DEFAULT_LOCALE = 'en';
+const DEFAULT_DATE_FORMAT = 'MM-dd-yyyy';
+
+const CustomInput = forwardRef((props: any, ref) => {
+  return <Input {...props} ref={ref} />;
+});
+
+interface DatePickerProps {
+  onChange: (data: Date | null) => void;
+  onBlur: () => void;
+  onFocus: () => void;
+  headerNodes: ReactNode;
+  disabled: boolean;
+  shouldCloseOnSelect: boolean;
+  fixedHeight: boolean;
+  startDate: Date | null;
+  endDate: Date | null;
+  customClassName: string;
+  popperClassName: string;
+  calendarClassName: string;
+  customTimeInput: ReactNode;
+  language: string;
+  yearsOptions: number[];
+  placeholder: string;
+  dateFormat: string;
+  selects: 'start' | 'end' | string;
+  selected: Date | null;
+}
+
+export const DatePicker: FC<DatePickerProps> = ({
+  onChange = () => {},
+  disabled = false,
+  onBlur = () => {},
+  onFocus = () => {},
+  endDate = null,
+  startDate = null,
+  headerNodes = null,
+  customClassName = '',
+  customTimeInput,
+  shouldCloseOnSelect = true,
+  popperClassName = '',
+  calendarClassName = '',
+  fixedHeight = false,
+  language = DEFAULT_LOCALE,
+  yearsOptions = [],
+  placeholder = DEFAULT_DATE_FORMAT.toUpperCase(),
+  dateFormat = DEFAULT_DATE_FORMAT,
+  selects = '',
+  selected = null,
+}) => {
+  const inputRef = useRef(null);
+  const startDateToString = startDate?.toDateString();
+  const endDateToString = endDate?.toDateString();
+  const isValidEndDate = endDate && startDate && endDate > startDate;
+
+  useMemo(() => registerLocale(language, currentLanguageToLocale[language]), [language]);
```

Comment:
According to [docs](https://github.com/Hacker0x01/react-datepicker/blob/main/README.md#localization) looks like the all supported locales can be registered once and then used just used a strings within `DatePicker`.
May be export `registerLocale` function together with component to allow users manage it outside in case they are willing to use none default locale?